src/main/java/seedu/address/model/person/student/Student.java

```diff
@@ -51,12 +51,21 @@ public Industry getIndustry() {
         return null;
     }
 
-    public void incrementAttendance() {
-        attendance++;
-    }
+    /**
+     * Returns true if both students have the same student id.
+     * This defines a weaker notion of equality between two students.
+     */
+    @Override
+    public boolean isSamePerson(Person otherPerson) {
+        if (otherPerson == this) {
+            return true;
+        }
 
-    public int getAttendance() {
-        return attendance;
+        if (otherPerson instanceof Student) {
+            return otherPerson != null
+                    && otherPerson.getStudentId().equals(getStudentId());
+        }
+        return false;
     }
```

Comment:
Thanks for enforcing student uniqueness by Student ID. We will need to update student ID as the primary identifier to the UG